### ISSUE TITLE
Install fails using PIP...related to not using pegged versions of dependencies in setup.py

### ISSUE THREAD

**Issue by willogden** [smcl/pyonfido#11]
Hi,

When installing using PIP, we see the following error:

```
Searching for requests
    Downloading https://pypi.python.org/packages/80/4a/ec7b2aaab37f7131224bb1f46998523fd13eec3911cd7a2add300a3cacca/requests-mock-1.2.0.tar.gz#md5=b82ac67a62ce2ba4e2fc2559f76b73f8
    Best match: requests mock-1.2.0
    Processing requests-mock-1.2.0.tar.gz
    Writing /tmp/easy_install-iL6eBF/requests-mock-1.2.0/setup.cfg
    Running requests-mock-1.2.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-iL6eBF/requests-mock-1.2.0/egg-dist-tmp-WNLezS

    Installed /tmp/easy_install-iL6eBF/requests-mock-1.2.0/.eggs/pbr-1.10.0-py2.7.egg
    removing '/tmp/pip-build-mSNOAk/PyOnfido/.eggs/requests_mock-1.2.0-py2.7.egg' (and everything under it)
    creating /tmp/pip-build-mSNOAk/PyOnfido/.eggs/requests_mock-1.2.0-py2.7.egg
    Extracting requests_mock-1.2.0-py2.7.egg to /tmp/pip-build-mSNOAk/PyOnfido/.eggs

    Installed /tmp/pip-build-mSNOAk/PyOnfido/.eggs/requests_mock-1.2.0-py2.7.egg
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-mSNOAk/PyOnfido/setup.py", line 38, in <module>
        'unittest2'
      File "/usr/local/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/local/lib/python2.7/site-packages/setuptools/dist.py", line 317, in __init__
        self.fetch_build_eggs(attrs['setup_requires'])
      File "/usr/local/lib/python2.7/site-packages/setuptools/dist.py", line 372, in fetch_build_eggs
        replace_conflicting=True,
      File "/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 854, in resolve
        raise DistributionNotFound(req, requirers)
    pkg_resources.DistributionNotFound: The 'requests' distribution was not found and is required by the application
```

For some reason `requests` is being resolved to `requests mock-1.2.0`!! We forked and changed setup.py as follows (version of requests is arbitrary...just needs to be pegged):

```
    install_requires=[
        'requests==2.5.1',
        'requests_mock',
        'unittest2'
    ],
    setup_requires=[
        'requests==2.5.1',
        'requests_mock',
        'unittest2'
    ],
```

This fixed the issue for us. I'm sure not everyone is experiencing this issue; not sure what it is about our setup that is causing this to happen (did try on various versions of PIP/setuptools/etc with no difference)...but it's probably good practice to peg (at least to major versions) the dependencies in setup.py? Would be good if we could drop our fork.

**Comment by hjheath** [comment]
It looks like you are hitting [this](http://stackoverflow.com/questions/27497470/setuptools-finds-wrong-package-during-install). What version of pip and python are you using? I could not reproduce installing in a clean virtualenv (Python 3.5.1, pip 9.0.1).

I would be happy to merge your fork - please create a PR and I will take a look.

**Comment by willogden** [comment]
Yes looks like the issue. Using Python 2.7.11 and pip 9.0.1. We're not using virtualenv (as building within a clean Docker container) so wonder if that's related. Will raise a PR, but you may want to use the latest version of requests, as 2.5.1 is pretty old now - it's just the version we're building against at the present.

**Comment by willogden** [comment]
PR created https://github.com/onfido/pyonfido/pull/12

**Comment by hjheath** [comment]
Fixed in version 0.6 - please let me know if you hit it again.
